src/NServiceBus.RabbitMQ/Configuration/ConnectionStringParser.cs

```diff
@@ -73,6 +73,15 @@ public ConnectionConfiguration Parse(string connectionString)
                 throw new NotSupportedException(message);
             }
 
+            if (ContainsKey("UsePublisherConfirms"))
+            {
+                var message = "The 'UsePublisherConfirms' connection string option has been removed. Use 'EndpointConfiguration.UsePublisherConfirms' instead.";
+
+                Logger.Error(message);
+
+                throw new NotSupportedException(message);
```

Comment:
I think that you are right, however I would first check for all the keys and throw exception with all violations. As currently when one would have 2 deprecated options in connection string, he would run the app, learn about 1st parameter not being supported, fix it. Later run it again and now get an exception about the 2nd paramter. So maybe we sould check all first and later throw  message including all not supported parameters. 

But that could be a separate PR. What do you think?